library/extractor/src/main/java/com/google/android/exoplayer2/metadata/id3/TextInformationFrame.java

```diff
@@ -19,52 +19,77 @@
 
 import android.os.Parcel;
 import android.os.Parcelable;
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import com.google.android.exoplayer2.MediaMetadata;
 import com.google.android.exoplayer2.util.Util;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 /** Text information ID3 frame. */
 public final class TextInformationFrame extends Id3Frame {
+  private final static String MULTI_VALUE_DELIMITER = ", ";
 
   @Nullable public final String description;
+
+  /** @deprecated Use {@code values} instead. */
+  @Deprecated
   public final String value;
 
-  public TextInformationFrame(String id, @Nullable String description, String value) {
+  @NonNull
+  public final String[] values;
+
+  public TextInformationFrame(String id, @Nullable String description, @NonNull String[] values) {
     super(id);
     this.description = description;
-    this.value = value;
+    this.values = values;
+
+    if (values.length > 0) {
+      this.value = values[0];
+    } else {
+      this.value = null;
+    }
+  }
+
+  /** @deprecated Use {@code TextInformationFrame(String id, String description, String[] values} instead */
+  @Deprecated
+  public TextInformationFrame(String id, @Nullable String description, String value) {
+    this(id, description, new String[] {value } );
   }
 
   /* package */ TextInformationFrame(Parcel in) {
     super(castNonNull(in.readString()));
     description = in.readString();
-    value = castNonNull(in.readString());
+    values = in.createStringArray();
+    this.value = values[0];
   }
 
   @Override
   public void populateMediaMetadata(MediaMetadata.Builder builder) {
+    // Depending on the context this frame is in, we either take the first value of a multi-value
+    // frame because multiple values make no sense, or we join the values together with a comma
+    // when multiple values do make sense.
     switch (id) {
       case "TT2":
       case "TIT2":
-        builder.setTitle(value);
+        builder.setTitle(values[0]);
```

Comment:
I think for this index-access to be safe you need to transform an empty array in the constructor to a single-entry array containing an empty string. Probably better to make this line (and similar lines below) robust in the case of an empty values list?

Please add a test for the case of passing in an empty values array.

Reply:
You're right about this. In fact, it's actually more consistent with the rest of the metadata module to make `values == [""]` rather than make it an empty list. I'll make that change.

Reply:
Resolved in 157f1f0.